LINK/usr/bin/evmexport

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+
+assert_num_args () {
+  num_args_received=$1
+  num_args_expected=$2
+  if [ "$num_args_received" -ne "$num_args_expected" ]
+  then
+    echo "Error: Wrong number of arguments (expected=$num_args_expected received=$num_args_received)"
+    usage
+    exit 1
+  fi
+}
+
+assert_export_dir_exists () {
+  export_dir=$1
+  if [ ! -d "$export_dir" ]
+  then
+    echo "Export directory doesn't exist! [${export_dir}]"
+    exit 1
+  fi
+}
+
+op_alerts () {
+  assert_num_args $# 1
+  EXPORT_DIR=$(readlink -v -f "$1")
+  assert_export_dir_exists "$EXPORT_DIR"
+
+  pushd /var/www/miq/vmdb > /dev/null
+  bin/rake "evm:export:alerts[${EXPORT_DIR},keep_spaces=${KEEP_SPACES}]"
+  popd > /dev/null
+}
+
+op_alertprofiles () {
+  assert_num_args $# 1
+  EXPORT_DIR=$(readlink -v -f "$1")
+  assert_export_dir_exists "$EXPORT_DIR"
+
+  pushd /var/www/miq/vmdb > /dev/null
+  bin/rake "evm:export:alertsets[${EXPORT_DIR},keep_spaces=${KEEP_SPACES}]"
+  popd > /dev/null
+}
+
+contains () {
+  SEARCH_TERM=$1
+  shift
+  LIST=$@
+
+  for element in $LIST; do
+    if [[ "$SEARCH_TERM" == "$element" ]]; then
+      return 0
+    fi
+  done
+
+  return 1
+}
+
+parse_arguments () {
+  # initialize non-required arguments before the loop in case they are
+  # not present in the command line
+  KEEP_SPACES="false"
+
+  # http://stackoverflow.com/questions/402377/using-getopts-in-bash-shell-script-to-get-long-and-short-command-line-options
+  # http://stackoverflow.com/questions/14786984/best-way-to-parse-cmdline-args-bash
+  while [[ $# -ge 1 ]]
+  do
+    arg="$1"
+    case "$arg" in
+      -s|--keep-spaces)
+        KEEP_SPACES="true"
+        shift # past the argument
+        ;;
+      --)
+        shift # past the argument
+        break # stop parsing arguments
+        ;;
+      *)
+        # unknown option
+        break # stop parsing arguments
+        ;;
+    esac
+  done
+
+  # if there are any required arguments check for them here
+
+  # pass the shifted arguments back to the caller
+  remaining_args="$@"
+}
+
+parse_action () {
+  AVAILABLE_PARSERS=`compgen -A function | grep '^op_'`
+
+  obj_type="$1"
+  op_func="op_${obj_type}"
+  contains $op_func $AVAILABLE_PARSERS; VALID_OP=$?
```

Comment:
Object type is a positional argument to the script and is required. I can make it so that it has to be passed in the option format instead of a positional argument, but there still needs to be some way to check that the type of object the user wants to export is valid (i.e. alert**s** is a valid export object type, but alert isn't) which is what this line is doing.